tests/resource/test_resource.py

```diff
@@ -180,6 +191,223 @@ def test_ipython_display(self, r, rep, capsys):
         captured = capsys.readouterr()
         assert rep in captured.out
 
+    expected_results_add_series = (
+        Resources(
+            2,
+            6,
+            defaultdict(int, {"RZ": 2, "CNOT": 1, "RY": 2, "Hadamard": 1}),
+            defaultdict(int, {1: 5, 2: 1}),
+            3,
+            Shots(10),
+        ),
+        Resources(
+            5,
+            6,
+            defaultdict(int, {"RZ": 2, "CNOT": 1, "RY": 2, "Hadamard": 1}),
+            defaultdict(int, {1: 5, 2: 1}),
+            3,
+            Shots(10),
+        ),
+        Resources(
+            2,
+            9,
+            defaultdict(int, {"RZ": 2, "CNOT": 1, "RY": 2, "Hadamard": 2, "PauliZ": 2}),
+            defaultdict(int, {1: 8, 2: 1}),
+            6,
+            Shots((10, (50, 2), 10)),
+        ),
+        Resources(
+            4,
+            8,
+            defaultdict(int, {"RZ": 2, "CNOT": 2, "RY": 2, "Hadamard": 2}),
+            defaultdict(int, {1: 6, 2: 2}),
+            5,
+            Shots((100, 10)),
+        ),
+    )
+
+    @pytest.mark.parametrize(
+        "resource_obj, expected_res_obj", zip(resource_quantities, expected_results_add_series)
+    )
+    def test_add_in_series(self, resource_obj, expected_res_obj):
+        """Test the add_in_series function works with Resoruces"""
+        other_obj = Resources(
+            num_wires=2,
+            num_gates=6,
+            gate_types=defaultdict(int, {"RZ": 2, "CNOT": 1, "RY": 2, "Hadamard": 1}),
+            gate_sizes=defaultdict(int, {1: 5, 2: 1}),
+            depth=3,
+            shots=Shots(10),
+        )
+
+        resultant_obj = add_in_series(resource_obj, other_obj)
+        assert resultant_obj == expected_res_obj
+
+        resultant_obj = resource_obj + other_obj
+        assert resultant_obj == expected_res_obj
```

Comment:
This tests a different function __add__. Fine to keep here but please consider an alternative implementation that each unit test targets a specific function.